src/main.functional.ts

```diff
@@ -0,0 +1,19 @@
+import { expect } from 'chai';
+import { app } from './main';
+
+describe('functional tests', () => {
+  /*eslint-disable */
+  //there are no types-supertest, so we use require.
+  //this throws lint error.
+  const request = require('supertest');
```

Comment:
We can't use the `import` syntax? e.g `import { request } from 'supertest' `

Reply:
what about using nock?

Reply:
ya types/superagent might work with superagent, not supertest right? 
will change to either of this package and fix circle ci, and put for review again - changing PR to wip

Reply:
found a @types/supertest, used it and we can import request now. https://www.npmjs.com/package/@types/supertest